cmd/import.go

```diff
@@ -74,7 +79,66 @@ func evolveSchema(ctx context.Context, db string, coll string, docs []json.RawMe
 	return util.Error(err, "create or update collection")
 }
 
+func fixNumbers(ctx context.Context, coll string, docs []json.RawMessage) {
+	err := schema.Infer(&sch, coll, docs[0:1], PrimaryKey, AutoGenerate, 1)
+	util.Fatal(err, "infer schema")
+
+	if schema.HasArrayOfObjects {
+		b, err := json.Marshal(schema.DummyRecord)
+		util.Fatal(err, "marshal number fixer record")
+
+		_, err = client.GetDB().Insert(ctx, coll, []driver.Document{b})
+		util.Fatal(err, "insert")
+
+		_, err = client.GetDB().Delete(ctx, coll, driver.Filter("{}"))
+		util.Fatal(err, "delete")
+	}
+}
+
+func guaranteeFloatsInFirstRecord(ctx context.Context, coll string, docs []json.RawMessage) {
+	if !FirstRecord {
+		return
+	}
+
+	cnt, err := client.GetDB().Count(ctx, coll, driver.Filter("{}"))
+
+	var ep *driver.Error
+
+	if err != nil {
+		if errors.As(err, &ep) && ep.Code == api.Code_NOT_FOUND {
+			log.Debug().Msg("collection doesn't exits, skip fixing numbers")
+			return
+		}
+
+		util.Fatal(err, "get count")
+	}
+
+	if cnt == 0 {
+		schema.DetectArrayOfObjects = true
+		schema.DetectIntegers = false
+		schema.ReplaceNumber = true
+
+		fixNumbers(ctx, coll, docs)
+
+		schema.DetectArrayOfObjects = false
+		schema.DetectIntegers = true
+		schema.ReplaceNumber = false
+
+		cnt, err := client.GetDB().Count(ctx, coll, driver.Filter("{}"))
+		util.Fatal(err, "get count after")
+
+		if cnt != 0 {
```

Comment:
Same thing here, all this fixing has to be transparent.

Reply:
Nothing is shown, only if there is error or debug log enabled.